object.go

```diff
@@ -151,7 +151,7 @@ func (o *Object) StoreProperty(p string, value interface{}) error {
 
 // SetProperty calls org.freedesktop.DBus.Properties.Set on the given
 // object. The property name must be given in interface.member notation.
-func (o *Object) SetProperty(p string, v interface{}) error {
```

Comment:
Can we fix this in a compatible way? For instance, can we do ParseVariant if we are passed something other than a variant before call?

Reply:
Sorry, I don't quite follow.

Currently, passing anything that's _not_ a variant fails; only a variant works.

Reply:
The issue is that changing the function signature will break the API which means some consumers will need to update their code, just trying to avoid those issues.

Reply:
Do you have an example call that works **and** would require changes? I just don't see it :sweat:

Reply:
See: https://github.com/coreos/go-systemd/pull/296 for a problem that occurred due to a previous change like this.

Reply:
This kind of issue occurs when someone is referencing the function as a value in their code, the types won't match.

Reply:
Sorry, I don't think we're understanding each other completely.

This function's current signature states that it receives a `v interface{}`. However, anything that is not an a `v Variant` will crash.

There are currently two types of applications using this function:

- Those passing a `Variant`. Those will continue to work.
- Those passing a non-`Variant`. Those do not work and crash right now. So while the API change would require them to make changes, it would require them to make changes to code that crashed in the first place.

So what I mean was, do you have any example code that _does_ work not, but would break with this change?

Reply:
You are correct, the behavior is currently broken. The compatibility issue is if someone is referring to the function as a value changing its signature will cause their code to no longer compile, as was the case in the issue linked above. Making the change as currently done in this PR would require a v6 release of the godbus package API. Since we generally try to avoid these kind of breaking changes, what I'm asking you to do is make passing an interface work correctly instead of changing the function API. In my original comment I suggested having the interface converted to a variant if it wasn't already one when passed. Is this not possible?

Reply:
Ah, now I get your point!

I think the scenario for possible breakage is a bit of an edge case, and might not exist, but your proposed fix has better results anyway, so I'll go ahead and try to implement that.

Reply:
Yes it is an edge case but it is still an API change which means that if we follow the general guidelines for go modules we need to increment the version number so we don't arbitrarily break user code. As we've seen before the type of change introduced here can break user code (whether or not this one actually would can't be said as we don't know how everyone is using the library). Backwards compatibility is a pain but unless one designs a better API for the library generally it is best keep individual functions API compatible. In other words we want to avoid a "github.com/godbus/dbus/v6" library until such a time as we are making a significant intentional change to the API.